indexer/database/bridge_transfers.go

```diff
@@ -230,25 +238,37 @@ func (db *bridgeTransfersDB) L2BridgeWithdrawalsByAddress(address common.Address
-
-	if cursor != "" {
-		withdrawalsQuery = withdrawalsQuery.Where("l2_bridge_withdrawals.id < ?", cursor)
-	}
-
-	filteredQuery := withdrawalsQuery.Where(&Transaction{FromAddress: address}).Order("l2_bridge_withdrawals.timestamp DESC").Limit(limit + 1)
-
+	// TODO join with l1_tokens and l2_tokens
+	ethAddressString := predeploys.LegacyERC20ETHAddr.String()
+
+	// Coalesce l2 transaction withdrawals that are simply ETH sends
+	ethTransactionWithdrawals := db.gorm.Model(&L2TransactionWithdrawal{})
+	ethTransactionWithdrawals = ethTransactionWithdrawals.Where(Transaction{FromAddress: address}).Where(`data = '0x' AND amount > 0`)
+	ethTransactionWithdrawals = ethTransactionWithdrawals.Joins("INNER JOIN l2_contract_events ON l2_contract_events.guid = l2_transaction_withdrawals.initiated_l2_event_guid")
+	ethTransactionWithdrawals = ethTransactionWithdrawals.Joins("LEFT JOIN l1_contract_events AS proven_l1_events ON proven_l1_events.guid = l2_transaction_withdrawals.proven_l1_event_guid")
+	ethTransactionWithdrawals = ethTransactionWithdrawals.Joins("LEFT JOIN l1_contract_events AS finalized_l1_events ON finalized_l1_events.guid = l2_transaction_withdrawals.finalized_l1_event_guid")
+	ethTransactionWithdrawals = ethTransactionWithdrawals.Select(`
+from_address, to_address, amount, data, withdrawal_hash AS transaction_withdrawal_hash,
+l2_contract_events.transaction_hash AS l2_transaction_hash, proven_l1_events.transaction_hash AS proven_l1_transaction_hash, finalized_l1_events.transaction_hash AS finalized_l1_transaction_hash,
+l2_transaction_withdrawals.timestamp, NULL AS cross_domain_message_hash, ? AS local_token_address, ? AS remote_token_address`, ethAddressString, ethAddressString)
+
+	withdrawalsQuery := db.gorm.Model(&L2BridgeWithdrawal{})
+	withdrawalsQuery = withdrawalsQuery.Joins("INNER JOIN l2_transaction_withdrawals ON withdrawal_hash = l2_bridge_withdrawals.transaction_withdrawal_hash")
+	withdrawalsQuery = withdrawalsQuery.Joins("INNER JOIN l2_contract_events ON l2_contract_events.guid = l2_transaction_withdrawals.initiated_l2_event_guid")
+	withdrawalsQuery = withdrawalsQuery.Joins("LEFT JOIN l1_contract_events AS proven_l1_events ON proven_l1_events.guid = l2_transaction_withdrawals.proven_l1_event_guid")
+	withdrawalsQuery = withdrawalsQuery.Joins("LEFT JOIN l1_contract_events AS finalized_l1_events ON finalized_l1_events.guid = l2_transaction_withdrawals.finalized_l1_event_guid")
+	withdrawalsQuery = withdrawalsQuery.Select(`
+l2_bridge_withdrawals.from_address, l2_bridge_withdrawals.to_address, l2_bridge_withdrawals.amount, l2_bridge_withdrawals.data, transaction_withdrawal_hash,
+l2_contract_events.transaction_hash AS l2_transaction_hash, proven_l1_events.transaction_hash AS proven_l1_transaction_hash, finalized_l1_events.transaction_hash AS finalized_l1_transaction_hash,
+l2_bridge_withdrawals.timestamp, cross_domain_message_hash, local_token_address, remote_token_address`)
+
+	// TODO: cursoring options
```

Comment:
Same doesn't the old cursoring logic work here?
```
	if cursor != "" {
		query = query.Where("l1_bridge_deposits.transaction_source_hash < ?", cursor)
	}

	query := query.Order("timestamp DESC").Limit(limit + 1)
```

Reply:
unrelated to pr I see I did 2 bugs in cursoring logic btw
1. I return all withdrawals instead of withdrawals-1
2. I don't check to see if withdrawals < limit +1

Reply:
ah i forgot to add back in the cursor block